go/spec/resolver.go

```diff
@@ -39,45 +39,42 @@ func (dsr *Resolver) resolveDatabaseString(str string) string {
 	return str
 }
 
-func (dsr *Resolver) resolvePathString(str string) string {
+func (dsr *Resolver) resolvePath(str string) string {
 	if dsr.config != nil {
 		split := strings.SplitN(str, separator, 2)
-		db, ds := "", ""
-		if len(split) == 1 {
-			// TODO: confirm that split[0] isn't a db spec
-			db = dsr.resolveDatabaseString("")
-			ds = split[0]
-		} else {
-			db = dsr.resolveDatabaseString(split[0])
-			ds = split[1]
+		db := ""
+		rest := split[0]
+		if len(split) > 1 {
+			db = split[0]
+			rest = split[1]
 		}
-		return db + separator + ds
+		return dsr.resolveDatabase(db) + separator + rest
 	}
 	return str
 }
 
 // Resolve string to database spec. If a config is present,
-//   - resolve a remote alias to its db spec
-//   - resolve "" to the local db spec
+//   - resolve a db alias to its db spec
+//   - resolve "" to the default db spec
 func (dsr *Resolver) GetDatabase(str string) (datas.Database, error) {
```

Comment:
I think you're right. All these cases are covered in the test but they're not run through the wrapper functions because require more setup. I think the clean way to handle this is to get rid of the wrappers and have the command line programs call the resolver first and that get the db, ds or object with a separate call.